src/views/homepage/components/tutorial-card/index.tsx

```diff
@@ -0,0 +1,74 @@
+import Link from 'next/link'
+import { VisuallyHidden } from '@reach/visually-hidden'
+import type { ProductSlug } from 'types/products'
+import { productSlugs } from 'lib/products'
+import ProductIcon from 'components/product-icon'
+import { IconPlay16 } from '@hashicorp/flight-icons/svg-react/play-16'
+import { IconTerminalScreen16 } from '@hashicorp/flight-icons/svg-react/terminal-screen-16'
+import Heading from 'components/heading'
+import Text from 'components/text'
+import s from './tutorial-card.module.css'
+
+const getProductIcons = () => {
+  const icons = {}
+  for (const slug of productSlugs) {
+    icons[slug] = <ProductIcon productSlug={slug} />
+  }
+  return icons
+}
+
+const BADGE_ICON_MAP = {
+  ...getProductIcons(),
+  video: <IconPlay16 />,
+  interactive: <IconTerminalScreen16 />,
+}
+
+interface TutorialCardProps {
+  link: string
+  duration?: string
+  heading: string
+  description: string
+  badges: Array<ProductSlug | 'video' | 'interactive'>
+}
+
+export default function TutorialCard({
+  link,
+  duration,
+  heading,
+  description,
+  badges,
+}: TutorialCardProps) {
+  return (
+    <Link href={link}>
```

Comment:
Can you use [the CardLink component](https://developer.hashi-mktg.com/swingset/components/cardlink) here?

Reply:
Is there a way to get the link to have a class? Looking to have equal height cards, but I don't have access to the wrapping link. 

<img width="813" alt="CleanShot 2022-04-13 at 12 50 09@2x" src="https://user-images.githubusercontent.com/825855/163230905-00269817-d821-4c41-be3b-f54d495893c6.png">

Reply:
We've done this using the `className` prop on `CardLink` before, `FeaturedTutorialsSection` is one place. They're located at the bottom of the downloads pages.

This is the class that gets applied to those cards, can you give something like this a try?

```css
.card {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  height: 100%;
}
```

Reply:
That works! Can this be a default as to not require consumers to add this? Feels like it'd be a common pattern to have a grouping of these cards that need to be of equal height?

Reply:
We don't be able to add it by default because there are several existing use cases where this wouldn't work, but I think we could either add a prop to `Card` that applies this class or we could create an additional `Card` component that's a variant of this. I'm leaning towards the prop idea since CSS is the only difference, maybe call it `isGridCard`? What do you think? Btw you aren't the first one to notice this pain, so totally valid issue to raise. 😊

Reply:
Do you have an example of where this would be problematic to be the default? Is it the 100% height declaration that makes it grow to large?

From my isolated testing adding a classname to the link with flex and adding flex to the root of card link grows the card as needed. Not extra CSS needed at the consumer level.

![CleanShot 2022-04-13 at 13 34 16@2x](https://user-images.githubusercontent.com/825855/163238204-af2663ae-74b1-41bc-a9a8-c2a48226a96b.png)

Reply:
Here are some places where `Card` is used, you can go ahead and try adding some default stuff now and testing it if you have time!

- The `InteractiveLabCallout` in the "Launch the lab environment" towards the bottom of this page: https://developer.hashi-mktg.com/vault/tutorials/app-integration/agent-read-secrets
- The Prev/Next cards at the bottom of learn content (can use same link as previous)
- The "This tutorial also appears in" section at the bottom of some pages: https://developer.hashi-mktg.com/vault/tutorials/app-integration/agent-kubernetes#help-and-reference
- The cards used on /vault/docs and /consul/docs (which can be tested by running `npm run start:vault` and `npm run start:consul`)
- Several cards located in the install pages, and cards within cards! https://developer.hashi-mktg.com/vault/downloads
- Product landing pages: https://developer.hashi-mktg.com/vault

Reply:
We ran into a similar issue in #178  - here's [a comment with some context](https://github.com/hashicorp/dev-portal/pull/178#discussion_r828156913).

At the time I'd proposed getting `Card` & `CardLink` to share the same CSS class name, rather than `CardLink` using the `Card` component. I think this could achieve the same styling consistency goal, without the extra wrapping element.

Maybe that's something we could revisit? I feel like this issue of having to add `height: 100%` is going to come up in a _ton_ of `Card` layouts 🤔

Reply:
I think `CardLink` should continue to use `Card`. I think it'd be great to find a CSS solution for this and I like Alex's proposal, I just want to make sure that the changes we make don't negatively impact other uses of `Card` and `CardLink` since they're not just used within grids.